swap/swap.go

```diff
@@ -128,37 +122,32 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	//load existing balances from the state store
+	// load existing balances from the state store
 	err = s.loadState(peer)
 	if err != nil && err != state.ErrNotFound {
 		log.Error("error while loading balance for peer", "peer", peer.ID().String())
 		return
 	}
 
-	//check if balance with peer is over the disconnect threshold
+	// check if balance with peer is over the disconnect threshold
 	if s.balances[peer.ID()] >= s.disconnectThreshold {
-		//if so, return error in order to abort the transfer
+		// if so, return error in order to abort the transfer
 		disconnectMessage := fmt.Sprintf("balance for peer %s is over the disconnect threshold %v, disconnecting", peer.ID().String(), s.disconnectThreshold)
 		log.Warn(disconnectMessage)
 		return errors.New(disconnectMessage)
 	}
 
-	//adjust the balance
-	//if amount is negative, it will decrease, otherwise increase
-	s.balances[peer.ID()] += amount
-
-	//save the new balance to the state store
-	peerBalance := s.balances[peer.ID()]
-	err = s.stateStore.Put(peer.ID().String(), &peerBalance)
+	// calculate new balance
+	var newBalance int64
+	newBalance, err = s.updateBalance(peer.ID(), amount)
 	if err != nil {
-		log.Error(fmt.Sprintf("error while storing balance for peer %s", peer.ID().String()))
-		return err
+		return
 	}
 
-	log.Debug(fmt.Sprintf("balance for peer %s after accounting: %s", peer.ID().String(), strconv.FormatInt(peerBalance, 10)))
-
-	//check if balance with peer is over the payment threshold
-	if peerBalance <= -s.paymentThreshold {
+	// check if balance with peer is over the payment threshold
```

Comment:
This comment should probably be removed in favor of the one 2 lines down.

Reply:
I changed it to `check if balance with peer crosses the threshold`